lib/puppet/functions/dns_a.rb

```diff
@@ -1,33 +1,17 @@
 # frozen_string_literal: true
 
-# Retrieves DNS A records and returns it as an array. Each record in the
-# array will be a IPv4 address.
-#
-# An optional lambda can be given to return a default value in case the
-# lookup fails. The lambda will only be called if the lookup failed.
+# [DEPRECATED] Retrieves DNS A records for a domain and returns them as an array.
```

Comment:
There's `@deprecated` according to https://www.rubydoc.info/gems/yard/file/docs/Tags.md#deprecated so perhaps that's cleaner. Not 100% sure it renders in the reference though so check it out.

Reply:
i have added the `@deprecated` tag but it dosen't seem to have an effect so i have left the [DEPRECATED] tag in place for now

Reply:
I don't see anything related in https://github.com/puppetlabs/puppet-strings/issues. Want to file an issue for it?

Reply:
https://github.com/puppetlabs/puppet-strings/issues/339